### PR TITLE
feat: Add MinMaxScale estimator, transformer & layer

### PR DESCRIPTION
**Pull request by georyetti**
### Description
Adds a min max scaling op in similar vein to the standard scaler

The below checklists come from the docs page on adding new transformers [here](../docs/adding_transformer.md)
### Keras Layer Checklist
Verify that:
- [x] The new Keras layer extends [BaseLayer](../src/kamae/tensorflow/layers/base.py)
- [x] The `_call` method has been implemented in the new layer.
- [x] The `compatible_dtypes` property is defined in the new layer.
- [x] The new layer is decorated with `@tf.keras.utils.register_keras_serializable(package=kamae.__name__)`.
- [x] The new layer takes a `name`, `input_dtype`, and `output_dtype` as arguments to the constructor and that this is passed to the super constructor.
- [x] The Keras layer is serializable. I have implemented the `get_config` method.
- [x] There are unit tests of the new layer. 
- [x] There is a specific test of layer serialisation added [here](../tests/kamae/tensorflow/test_layer_serialisation.py).
- [x] The new layer is imported in the [__init__.py](../src/kamae/tensorflow/layers/__init__.py) file in the `layers` directory.

### Spark Transformer/Estimator Checklist
Verify that:
- [x] The new Spark Transformer extends [BaseTransformer](../src/kamae/spark/transformers/base.py).
- [x] If the new transform needs a fit method, a Spark Estimator has been implemented that extends [BaseEstimator](../src/kamae/spark/estimators/base.py).
- [x] The instructions in the above docs page have been followed for the `__init__` and `setParams` methods.
- [x] The transformer uses one of the input/output mixin classes from [base.py](../src/kamae/spark/params/base.py).
- [x] If the new transformer requires more parameters that would need to be serialised to the Spark ML pipeline, there is a implemented parameter class by extending the `Params` class [here](https://spark.apache.org/docs/latest/api/python/reference/api/pyspark.ml.param.Params.html).
- [x] The `compatible_dtypes` property has been implemented to specify the input/output data types that my transformer/estimator supports.
- [x] A Keras subclassed layer is returned in the transformer's `get_tf_layer` method.
- [x] There are unit tests of the new transform. In particular, there are parity tests between the Spark and Keras implementations.
- [x] The new transformer/estimator is imported in the [__init__.py](../src/kamae/spark/transformers/__init__.py) file in the `transformers`/`estimators` directory.

Finally, please verify that:
- [x] There is a new entry (alphabetical order) in the README table describing the new layer/transformer

### CODE REVIEW

File: tests/kamae/spark/transformers/test_min_max_scale.py [hunk 1]
```diff
@@ -0,0 +1,450 @@
+# Copyright [2024] Expedia, Inc.
```
Comment (jacobjwood) [inline tests/kamae/spark/transformers/test_min_max_scale.py:1]: Maybe a trivial case, but should we add a test to check that the None checks for min/max values does throw an exception? I saw it was missing in the coverage report

File: src/kamae/tensorflow/layers/min_max_scale.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        the input is a single tensor. Raises an error if multiple tensors are passed
+        in as an iterable.
+        :param inputs: Input tensor to perform the normalization on.
+        :returns: The input tensor with the normalization applied.
+        """
+        # Ensure min and max match input dtype.
+        min_tensor = self._cast(self.min, inputs.dtype.name)
+        max_tensor = self._cast(self.max, inputs.dtype.name)
+        normalized_outputs = tf.math.divide_no_nan(
+            tf.math.subtract(inputs, min_tensor), max_tensor - min_tensor
```
Comment (jacobjwood) [inline src/kamae/tensorflow/layers/min_max_scale.py:175]: is there a reason for using tf.math.subtract in the first arg and using - in the second?

**Review by jacobjwood — COMMENTED** [review]
On the whole looks good - some minor comments

File: src/kamae/tensorflow/layers/min_max_scale.py (hunk 2 above)
Comment (georyetti) [reply src/kamae/tensorflow/layers/min_max_scale.py:175]: No will align the two

File: src/kamae/spark/estimators/min_max_scale.py [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+        """
+        input_column_type = self.get_column_datatype(dataset, self.getInputCol())
+        if not isinstance(input_column_type, ArrayType):
+            input_col = F.array(F.col(self.getInputCol()))
+            input_column_type = ArrayType(input_column_type)
+        else:
+            input_col = F.col(self.getInputCol())
+
+        # Collect a single row to driver and get the length.
+        # We assume all subsequent rows have the same length.
```
[the remark this answers is not in the document]
Comment (georyetti) [reply src/kamae/spark/estimators/min_max_scale.py:103]: Will add to all the scalers now

File: src/kamae/tensorflow/layers/min_max_scale.py [hunk 4]
```diff
@@ -0,0 +1,182 @@
...
+        the input is a single tensor. Raises an error if multiple tensors are passed
+        in as an iterable.
+        :param inputs: Input tensor to perform the normalization on.
+        :returns: The input tensor with the normalization applied.
+        """
+        # Ensure min and max match input dtype.
+        min_tensor = self._cast(self.min, inputs.dtype.name)
+        max_tensor = self._cast(self.max, inputs.dtype.name)
+        normalized_outputs = tf.math.divide_no_nan(
+            tf.math.subtract(inputs, min_tensor), tf.subtract(max_tensor, min_tensor)
```
Comment (jacobjwood) [inline src/kamae/tensorflow/layers/min_max_scale.py:175]: Nit: but one here uses math.subtract and the other uses tf.subtract. Does it need to be consistent? Given tests have passed it probably works fine, but should consider changing this

**Review by jacobjwood — APPROVED** [review]
A minor nit, but don't believe it affects functionality so LGTM

**Review by jacobjwood — APPROVED** [review]
